src/client/Microsoft.Identity.Client/Platforms/Android/AndroidBroker.cs

```diff
@@ -96,7 +96,6 @@ private async Task AcquireTokenInternalAsync(IDictionary<string, string> brokerP
                     _logger.Verbose("User is specified for silent token request. Starting silent broker request");
                     string silentResult = await _brokerHelper.GetBrokerAuthTokenSilentlyAsync(brokerPayload, _activity).ConfigureAwait(false);
                     s_androidBrokerTokenResponse = CreateMsalTokenResponseFromResult(silentResult);
-                    s_readyForResponse?.Release();
                     return;
```

Comment:
@jennyf19 - it took me a while to figure out what's going on. Suffice to say that we use a semaphore like a lock. If you call "wait", the lock becomes "busy". If you call "release", it becomes available. Waiting on a busy lock "blocks" until the lock becomes available again. The semaphore starts as "busy".

When doing a Silent auth, there is no need really for a lock, because the broker just returns the result. So the bug was that the lock got released, but there's nothing waiting on that lock. A subsequent interactive request would fail because at line 142 (below) would simply continue, since the lock is not busy.

`await s_readyForResponse.WaitAsync()`

Reply:
All this manual locking is very tedious and bug prone. Tasks are a much better abstraction that can be used to do multi-threading and asyncronous.

Reply:
i see. thanks for the explanation @bgavrilMS

Reply:
you are exactly right @bgavrilMS